hazelcast/src/test/java/com/hazelcast/cache/CachePartitionIteratorBouncingTest.java

```diff
@@ -0,0 +1,184 @@
+            testTasks[i++] = new IterationRunnable(driver);
+            testTasks[i++] = new MutationRunnable(driver, i / 2);
+        }
+        bounceMemberRule.testRepeatedly(testTasks, MINUTES.toSeconds(3));
+    }
+
+    private void populateCache(ICache<Integer, Integer> cache) {
+        for (int i = 0; i < STABLE_ENTRY_COUNT; i++) {
+            cache.put(i, i);
+        }
+    }
+
+    public class IterationRunnable implements Runnable {
+
+        private final HazelcastInstance hazelcastInstance;
+        private ICache<Integer, Integer> cache;
+
+        public IterationRunnable(HazelcastInstance hazelcastInstance) {
+            this.hazelcastInstance = hazelcastInstance;
+        }
+
+        @Override
+        public void run() {
+            if (cache == null) {
+                cache = hazelcastInstance.getCacheManager().getCache(TEST_CACHE_NAME);
+            }
+            HashSet<Integer> all = getAll();
+            for (int i = 0; i < STABLE_ENTRY_COUNT; i++) {
+                assertTrue("Missing stable entry " + i + " - " + cache.get(i), all.contains(i));
+            }
+
+            logger.info("Successfully finished iteration " + successfulIterations.incrementAndGet());
+        }
+
+        private HashSet<Integer> getAll() {
+            HashSet<Integer> keys = new HashSet<>();
+            int partitionCount = hazelcastInstance.getPartitionService().getPartitions().size();
+            for (int partitionId = 0; partitionId < partitionCount; partitionId++) {
+                Iterator<Cache.Entry<Integer, Integer>> iterator = createIterator(cache, FETCH_SIZE, partitionId, false);
+                while (iterator.hasNext()) {
+                    Entry<Integer, Integer> e = iterator.next();
+                    assertTrue("Got the same key twice", keys.add(e.getKey()));
+                }
+            }
+            return keys;
+        }
+
+    }
+
+    private Iterator<Cache.Entry<Integer, Integer>> createIterator(
+            ICache<Integer, Integer> cache, int fetchSize, int partitionId, boolean prefetchValues) {
+        return isClientDriver()
+                ? ((ClientCacheProxy<Integer, Integer>) cache).iterator(fetchSize, partitionId, prefetchValues)
+                : ((CacheProxy<Integer, Integer>) cache).iterator(fetchSize, partitionId, prefetchValues);
+    }
+
+    protected boolean isClientDriver() {
+        return false;
+    }
+
+    protected InMemoryFormat getInMemoryFormat() {
+        return CacheSimpleConfig.DEFAULT_IN_MEMORY_FORMAT;
+    }
+
+    public static class MutationRunnable implements Runnable {
+        private final HazelcastInstance hazelcastInstance;
+        private final int startIndex;
+        private final int endIndex;
+        private ICache<Integer, Integer> cache;
+
+        public MutationRunnable(HazelcastInstance hazelcastInstance, int runnableIndex) {
+            this.hazelcastInstance = hazelcastInstance;
```

Comment:
We call the method twice with `runnableIndex` = 0 in both cases? Is this an idea of the test?

Reply:
Which method and which cases? As the for-loop increases by two in each loop, the iteration index should be increasing by one.

Reply:
We call it twice here and `i/2` gives 0. 

```
        for (int i = 0; i < CONCURRENCY; ) {
            HazelcastInstance driver = bounceMemberRule.getNextTestDriver();
            testTasks[i++] = new IterationRunnable(driver);
            testTasks[i++] = new MutationRunnable(driver, i / 2);
        }
```

Reply:
Didn't notice that it increases by 2.